Makefile

```diff
@@ -10,8 +10,10 @@ SHELL=/bin/bash -o pipefail
 
 PROJECT_NAME=bookkeeper-operator
 REPO=pravega/$(PROJECT_NAME)
-VERSION=$(shell git describe --always --tags --dirty | sed "s/\(.*\)-g`git rev-parse --short HEAD`/\1/")
+BASE_VERSION=0.1.6
+ID=$(shell git rev-list `git rev-list --tags --no-walk --max-count=1`..HEAD --count)
```

Comment:
any particular reason to count commits between HEAD and last tag commit?

As a general rule of thumb the total number of commits from the current HEAD is used as a build number .

Reply:
I had included it in this fashion since counting commits between HEAD and last tag commit would give a sense of the number of commits that have gone in since the last release, and a combination of the fields `{base_version}-{number_of_commits_after_last_release}-{commit_id}` would always generate a unique build number.
But I could modify this and use the total number of commits from the current HEAD as well if that is the rule that is generally followed.

Reply:
Counting number of commits from the current HEAD gives you unique number as well and also provides sense of total number of changes across versions. Also all commits from previous releases are included in the next one is dropping them from the build number doesn't look good.

Reply:
fixed it